### ISSUE TITLE
Fix false positives for .toLocaleString() options argument in property-no-unknown

### ISSUE THREAD

**Issue by SimonSimCity** [stylelint/stylelint#3807]
When checking the following javascript code:

```js
(positionsChecked / scansCount).toLocaleString('de-DE', { style: 'percent' })
```

I get the following feedback from stylelint:

> Unexpected unknown property "style" (property-no-unknown)

I've experienced this bug in the visual-studio plugin [vscode-stylelint](https://github.com/shinnn/vscode-stylelint) and they said this issue should be reported here: https://github.com/shinnn/vscode-stylelint/issues/253

**Comment by jeddy3** [comment]
@SimonSimCity Thanks for the report. This appears to be an issue with the parser stylelint is using for styled components. Can you raise an issue upstream in the [`postcss-styled` issue tracker](https://github.com/gucong3000/postcss-styled/issues), please?

As a workaround until the issue is fixed upstream, you can use the following in your stylelint config object:

```json
{
  "rules": {
    "property-no-unknown": [true, { "ignoreProperties": ["style"] }]
  }
}
```

**Comment by SimonSimCity** [comment]
@jeddy3 Well, I'd like if you would do it because I don't know how you interfere with the named library. I can just copy this issue to the issue tracker of `postcss-styled` but I don't know if they will understand it right away.

**Comment by jeddy3** [comment]
No worries. I see you've been bounced from another issue tracker to ours already.

@gucong3000 Should I report this in `postcss-styled` or `postcss-jsx`?

**Comment by gucong3000** [comment]
@SimonSimCity 
```
(positionsChecked / scansCount).toLocaleString('de-DE', { style: 'percent' })
```
Is there any filename in vscode?
`vscode-stylelint` cannot completely correctly judge matching parser until it is saved as the correct file name

**Comment by SimonSimCity** [comment]
@gucong3000 I've saved it as a `.js` file.

**Comment by SimonSimCity** [comment]
Disabling only `property-no-unknown` would be a better way of doing this - but it's a work-around and doesn't _solve_ the problem.

**Comment by jeddy3** [comment]
> I don't have any way to recreate this problem.

Thanks for looking into it and adding the test.

I'm going to close for now. @SimonSimCity Hopefully it'll start working for you, otherwise please continue to use the workaround.

**Comment by SimonSimCity** [comment]
Surprising to me is that it isn't marked as error when using inside a script-tag in an HTML file. In this sample, I don't even get an error - just a warning - but then also only on the two properties that really do not exist.

```
<html style="base: false;">
  <body style="test: fpp; color: black;">

    <script>
      const data = {};

      export default [
        { text: `${(data.price).toLocaleString('de-DE', { style: 'currency', currency: 'EUR' })}`, fontSize: 8, bold: false, alignment: 'right' },
        { text: `${(data.price * data.count).toLocaleString('de-DE', { style: 'currency', currency: 'EUR' })}`, fontSize: 8, bold: false, alignment: 'right' },
      ];

      (data.scan / data.price).toLocaleString('de-DE', { style: 'percent' })
    </script>
  </body>
</html>
```

**Comment by jeddy3** [comment]
> Please try again with the following setting:

@SimonSimCity Thanks for the follow-up. I'm able to reproduce the violation when following your instructions. I'll reopen the issue.

@gucong3000 `./node_modules/.bin/stylelint "sources/js/test.js" --syntax styled` gives _no_ violation, whereas `./node_modules/.bin/stylelint "sources/js/test.js" --syntax jsx` does. So it might be due to the heuristics  in `postcss-css-in-js` (aka `postcss-jsx`).

---

> Surprising to me is that it isn't marked as error when using inside a script-tag in an HTML file

I don't think we've ever considered this use case i.e. supporting CSS-in-JS within a `<script>`  tag. @SimonSimCity Please can you create a new [feature request](https://github.com/stylelint/stylelint/issues/new?template=REQUEST_A_FEATURE.md) issue for this.

**Comment by gucong3000** [comment]
I think I found the reason (`supports[nameSpace[0]]`) for the error, it is here:
https://github.com/gucong3000/postcss-jsx/blob/5f08a45ad4b09354691a69e20ff58bb567a72398/extract.js#L198


But it has been fixed in v0.36.0, so updating to the latest version of stylelint can fix this problem.

**Comment by jeddy3** [comment]
@gucong3000 Thanks for looking into this again.

I can confirm this is no longer reproducible using [the demo](https://stylelint.io/demo/).
